python/plugins/processing/tools/postgis.py

```diff
@@ -59,6 +59,9 @@ def uri_from_name(conn_name):
 
     settings.endGroup()
 
+    if hasattr(authcfg, 'isNull') and authcfg.isNull():
```

Comment:
The problem is that QSettings do not return a NULL QString, but a NULL QVariant : 
```
arguments did not match any overloaded call: QgsDataSourceUri.setConnection(str, str, str, str, str, QgsDataSourceUri.SslMode sslmode=QgsDataSourceUri.SslPrefer, str authConfigId=QString()): argument 7 has unexpected type 'QVariant' QgsDataSourceUri.setConnection(str, str, str, str, QgsDataSourceUri.SslMode sslmode=QgsDataSourceUri.SslPrefer, str authConfigId=QString()): argument 5 has unexpected type 'str' See log for more details
```